### ISSUE TITLE
Determine Ruby and Puppet version support

### ISSUE THREAD

**Issue by rnelson0** [voxpupuli/hiera-eyaml#238]
We should discuss what versions of Ruby and Puppet hiera-eyaml should support, then look at what versions it supports now and how we get from A to B.

My thoughts:

First up is Ruby 1.8.7. We are experiencing issues with the tests, and it's no longer supported by Puppet or Vox Pupuli. I think we should remove this support. 

Second is Ruby 1.9.3. No tests are currently failing but it is no longer supported by Puppet (company) or VP. However, Puppet (the product) 3 is still supported by this gem which itself supports Ruby 1.9.3. I think this should be phased out at the same time as puppet 3.

Third would be Ruby 2.0. Tests are not failing, VP generally does not test with it, and I am not sure of Puppet's support for it. I think we should remove it for standardization with other VP tooling. 

These can be removed together, in a single major version bump. 

This leaves tidying up the specific point releases of Ruby 2.1/2.1 and Puppet 4 that are in use. We should true them up with what we test puppet modules with.

In summary, I am suggesting two steps:

* Remove Ruby 1.8.7, 1.9.3, 2.0.0; Puppet 3.x; bump major version of hiera-eyaml
* True up Ruby 2.1, 2.2; Puppet 4.x versions; bump minor version
